pkg/alicloud/client/client.go

```diff
@@ -355,3 +358,78 @@ func (c *vpcClient) DescribeNatGateways(req *vpc.DescribeNatGatewaysRequest) (*v
 func (c *vpcClient) DescribeEipAddresses(req *vpc.DescribeEipAddressesRequest) (*vpc.DescribeEipAddressesResponse, error) {
 	return c.client.DescribeEipAddresses(req)
 }
+
+// vpcClient defines the struct of VPC client
+type ramClient struct {
+	client *ram.Client
+}
+
+// NewSTSClient creates a new STS client with given region, AccessKeyID, and AccessKeySecret
+func (f *clientFactory) NewRAMClient(region, accessKeyID, accessKeySecret string) (RAM, error) {
+	client, err := ram.NewClientWithAccessKey(region, accessKeyID, accessKeySecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &ramClient{
+		client: client,
+	}, nil
+}
+
+func (c *ramClient) GetServiceLinkedRole(regionID, roleName string) (*ram.Role, error) {
+	request := ram.CreateGetRoleRequest()
+	request.RoleName = roleName
+	request.SetScheme("HTTPS")
+	request.RegionId = regionID
+
+	response, err := c.client.GetRole(request)
+	if err != nil {
+		if c.IsNoPermissionError(err) {
+			return nil, fmt.Errorf("no permission to get service linked role, please grant credentials correct privileges. See https://github.com/gardener/gardener-extension-provider-alicloud/blob/master/docs/usage-as-end-user.md#Permissions")
+		}
+		if c.IsRoleNotExistsError(err) {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	if !response.Role.IsServiceLinkedRole {
+		return nil, fmt.Errorf("%v is not service linked role", roleName)
```

Comment:
change `not service linked role` to `a not service linked role`

Reply:
done